app/validators/same_parent_oda_type_validator.rb

```diff
@@ -0,0 +1,11 @@
+class SameParentOdaTypeValidator < ActiveModel::Validator
+  def validate(activity)
+    return unless activity.is_ispf_funded?
+    return if activity.is_oda.nil?
+    return if activity.parent.fund?
```

Comment:
Could this be `activity.programme?` for clarity?

Reply:
Could I probe on what you mean by clarity here?

I went back and forth on this a little, and this is going to get a weirdly philosophical, but I'll do my best at trying to explain my thought process!

Although there's not _really_ a difference here in terms of the code logic (a `Programme`'s `parent` should always be a `fund` in the app's current state), I felt there was a distinction between an activity being a "Programme" and its parent being a "Fund". We're validating the parent in other places here, so it feels like, to me, _that's_ the entity with a value we want to check against our current entity (a "Programme", or anything below it).

If for any reason (highly unlikely) the hierarchy of levels were to change, and a "Project" was the top level entity, this would be _slightly_ more future-proof as well. I appreciate there are tons of other places this won't apply, so this may not be a compelling argument...

Hope that made some sort of sense, I've struggled to put my thoughts into words here!